### PR TITLE
Refactoring/Sad Path Testing

### PR DESCRIPTION
**Pull request by blakesaylor**
## PR for Issues/User Story #(s): 
  
- [ ] **Fixes**: 
> - 

- [x] **Completes**:
> - Completes #67, #68. 
> - UserSquads: Adds testing for index and destroy methods. 
> - Games: Added sad path testing for the Show method. The external API is super forgiving, so I created a test using a gigantic string to get an invalid response from the API. I believe that this is as good as we’re going to get since even searching the API using strings (e.g. https://api.rawg.io/api/games/pickle, https://api.rawg.io/api/games/test ) still returns results

- [x] **Updates**:
> - Updates fixture files based on changes to tests

- [x] **Needs attention**:  
> - UserSquads: The create method in here is currently not being tested as I don’t think it’s actually even being used. This is where the Mailer exists currently, I think we’re going to need to move that Mailer to a different method (api/v1/squads_controller.rb#create)

**Other Notes**:
> -

### CODE REVIEW

File: app/controllers/api/v1/games_controller.rb [hunk 1]
```diff
@@ -1,15 +1,10 @@
...
   def show
     game = RawgFacade.get_game(params[:id])
-    render json: game, status: 200
-  end
-
-  private
-  def param_validation
-    if !params[:id].present?
-      render json: { data: {} }, status:400
+    if !game.nil?
```
Comment (Bontgoy) [inline app/controllers/api/v1/games_controller.rb:4]: use `game.present?`

File: app/controllers/api/v1/games_controller.rb [hunk 2]
```diff
@@ -1,15 +1,10 @@
...
+      render json: game, status: 200
+    else
+      render json: { description: "Error: No game found" }, status: 404
```
Comment (Bontgoy) [inline app/controllers/api/v1/games_controller.rb:7]: little rails sugar here but this is a good option for readability

`status: :not_found`

File: app/controllers/api/v1/games_controller.rb (hunk 1 above)
Comment (Bontgoy) [reply app/controllers/api/v1/games_controller.rb:4]: It is always nice to try and avoid inverse conditions when you can. Its not as simple to understand plus rails has cool stuff like `.present?` that developers are super familiar with.

File: app/controllers/api/v1/users/squads_controller.rb [hunk 3]
```diff
@@ -1,8 +1,12 @@
 class Api::V1::Users::SquadsController < ApplicationController
   def index
-    user = User.find_by(params[:id])
-    squads = UserSquad.where(user_id: user.id)
-    render json: UserSquadSerializer.new(squads), status: 201
+    user = User.find_by(id: params[:user_id])
+    if !user.nil?
```
Comment (Bontgoy) [inline app/controllers/api/v1/users/squads_controller.rb:4]: `user.present?`

File: app/controllers/api/v1/users/squads_controller.rb [hunk 4]
```diff
@@ -12,7 +16,12 @@ def create
   end
 
   def destroy
-    user_squad = UserSquad.find(params[:id])
-    user_squad.destroy
+    user_squad = UserSquad.find_by(id: params[:id])
+    if !user_squad.nil?
```
Comment (Bontgoy) [inline app/controllers/api/v1/users/squads_controller.rb:20]: `.present?`

File: app/controllers/api/v1/users/squads_controller.rb [hunk 5]
```diff
@@ -1,8 +1,12 @@
 class Api::V1::Users::SquadsController < ApplicationController
   def index
-    user = User.find_by(params[:id])
-    squads = UserSquad.where(user_id: user.id)
-    render json: UserSquadSerializer.new(squads), status: 201
+    user = User.find_by(id: params[:user_id])
+    if !user.nil?
+      squads = UserSquad.where(user_id: user.id)
```
Comment (Bontgoy) [inline app/controllers/api/v1/users/squads_controller.rb:5]: If you have setup the `has_many` association in the model you can use `user.squads`

File: app/controllers/api/v1/users/squads_controller.rb [hunk 6]
```diff
@@ -12,7 +16,12 @@ def create
   end
 
   def destroy
-    user_squad = UserSquad.find(params[:id])
-    user_squad.destroy
+    user_squad = UserSquad.find_by(id: params[:id])
+    if !user_squad.nil?
+      user_squad.destroy
+      render json: { description: "UserSquad deleted successfully" }, status: 204
```
Comment (Bontgoy) [inline app/controllers/api/v1/users/squads_controller.rb:22]: On `destroy` it is very common to just return

```
render json: {}, status: :ok
```

If you're already consuming the message on the FE then just consider this a suggestion.

**Review by Bontgoy — APPROVED** [review]
Looks good - great testing! Mostly some sugar suggestions here. There are other best practices we could talk about like making a private user method in the base controller called like

```
def current_user
  @current_user ||= User.find(params[:id])
end
```

And also how to deal with not found records without `find_by` but that's just shed painting and things to think about as you dive deeper into your work.

:shipit:

File: app/controllers/api/v1/users/squads_controller.rb (hunk 5 above)
Comment (juliet-e) [reply app/controllers/api/v1/users/squads_controller.rb:5]: +1 - leverage AR associations

**Review by juliet-e — APPROVED** [review]

**Pull request merged**